core/kernel/dt.c

```diff
@@ -51,6 +51,99 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop = NULL;
+	int len_prop = 0;
+	int it_num = DT_INFO_INVALID_INTERRUPT;
+
+	/*
+	 * Interrupt property can be defined with at least 2x32 bits word
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if (!int_prop || (len_prop < 2))
+		return it_num;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+int dt_find_node_by_compatible_and_status(void *fdt, int status, int node,
+						const char *compat)
+{
+	int rnode = -FDT_ERR_NOTFOUND;
+	int fnode = -FDT_ERR_NOTFOUND;
+	int node_status = 0;
+
+	fnode = fdt_node_offset_by_compatible(fdt, node, compat);
+
+	while (fnode != -FDT_ERR_NOTFOUND) {
+		node_status = _fdt_get_status(fdt, fnode);
+		if (node_status & status) {
+			rnode = fnode;
+			break;
+		}
+
+		fnode = fdt_node_offset_by_compatible(fdt, fnode, compat);
+	}
+
+	return rnode;
+}
+
+int dt_disable_status(void *fdt, int node)
+{
+	const char *prop = NULL;
+	int len = 0;
+
+	prop = fdt_getprop(fdt, node, "status", &len);
+	if (!prop) {
+		/* Status is not available, just add it */
+		if (fdt_setprop_string(fdt, node, "status", "disabled"))
+			return -1;
+	} else {
+		/*
+		 * Status is there, modify it.
+		 * Ask to set "disabled" value to the property. The value
+		 * will be automatically truncated with "len" size by the
+		 * fdt_setprop_inplace function.
+		 * Setting a value different from "ok" or "okay" will disable
+		 * the property.
+		 * Setting a truncated value of "disabled" with the original
+		 * property "len" is preferred to not increase the DT size and
+		 * losing time in recalculating the overall DT offsets.
+		 * If original length of the status property is larger than
+		 * "disabled", the property will start with "disabled" and be
+		 * completed with the rest of the original property.
+		 */
+		if (fdt_setprop_inplace(fdt, node, "status", "disabled", len))
+			return -1;
+	}
+
+	return 0;
+}
+
+int dt_enable_secure_status(void *fdt, int node)
+{
+	if (dt_disable_status(fdt, node)) {
+		EMSG("Unable to disable Normal Status");
+		return -1;
+	}
+
+	/*
+	 * Create or change the "secure-status" property in the DT.
+	 * If the property is not present, it will be added
+	 * otherwise it will be modified to be "okay"
+	 */
+	if (fdt_setprop_string(fdt, node, "secure-status", "okay"))
```

Comment:
what if secure-status already exists?
This should do the same trick as done in `dt_disable_status()`. Even a bit more. Case string is "d\0", there's no room to change in place into "ok\0".

Reply:
if secure-status already exists the value will be forced to `okay`. We can not apply the same as dt_disable_status because to disable a status, writing something different of `ok` or `okay` is doing the job. Here we want to enable the status so we have to set it explicitly to `ok` or `okay`. Choice has been done to set it to `okay`. (refer to `is_okay()` in the same file.

Reply:
Ok, fine for setting `secure-status` to `okay`. 
That said, you still face with `secure-status` the same problematic as faced with `status`. If `secure-status` is already defined but shorter than 6 chars (including `\0`) you must take care not to override DT data by loading string "okay".

Reply:
For me this is not obvious. The `fdt_setprop_string` is not supposed to do the job for us? In the `fdt_setprop_string` description it's said "create or change a property" and "This function may insert or delete data from the blob, and will therefore change the offsets of some existing nodes"
So what do you want me to do more. Do you want me the check if the `secure-status` already exist, then I have to delete it and recreate it?

Reply:
If `fdt_setprop_string(fdt, node, "secure-status", "okay")` is fine to force "secure-status" to "okay", then you should be able to implement `dt_disable_status()` the same way:
```C
int dt_disable_status(void *fdt, int node)
{
	if (fdt_setprop_string(fdt, node, "status", "disabled"))
		return -1;

 	return 0;
}
```

Reply:
Yes, but do we want to spend time in recalculating all the Device Tree when the `status` property is already present and just changing the `ok` or `okay` with `di` or `disab` is disabling it without recalculating all the device tree?
Let me know.

Reply:
I get your point. I still think it's a bit strange to see those 2 functions with such different implementation whereas they have about the same purpose.

Minor: can remove the inline comment in `dt_enable_secure_status()`, it adds no info.